numpy/polynomial/_polybase.py

```diff
@@ -810,6 +813,60 @@ class domain in NumPy 1.4 and ``None`` in later versions.
             return cls(coef, domain=domain, window=window)
 
     @classmethod
+    def interp(cls, func, deg, x=None, args=(), window=None):
+        """Approximate a function using polynomial interpolation.
+
+        Return the series instance that interpolates a given function.
+
+        Parameters
+        ----------
+        func : function
+            The function to be approximated. It must be a function of a single
+            variable of the form f(x,a,b,c...), where a,b,c... are extra arguments
+            that can be passed in the `args` parameter.
+        deg : int or 1-D array_like
+            Degree(s) of the fitting polynomials. All terms up to and including the
+            `deg`'th term are included in the interpolation.
+        x : array_like, shape (M,), optional
+            x-coordinates of the M interpolation points. The window of the series
+            is set by the extrema of this list if `window` is not set.
+        args : tuple, optional
+            Extra arguments to be used in the function call.
+        domain : [beg, end], optional
+            Domain to use for the returned series.
+        window : [beg, end], optional
+            Window to use for the returned series. The default value is the
```

Comment:
The `domain` is the range of x over which you want to fit the function, the window is where the `domain` is mapped for the fit, and also where the polynomials are evaluated for the y values. For the Legendre and Chebyshev functions the window should almost always be `[-1, 1]`, especially for fitting, because that is where the weight is defined and the special properties hold. The `window` parameter was introduced for the Hermite and Laguerre functions because of their weight functions did not have compact support. Not that I think it has been all that successful, I'd probably do it with type specific scale and offset if I had to do it again.